### ISSUE TITLE
invalid memory address or nil pointer dereference in agent

### ISSUE THREAD

**Issue by rmporsch** [kubeflow/kfserving#1317]
/kind bug

**What steps did you take and what happened:**

We deployed the following triton inferenceservice:

```yaml
apiVersion: "serving.kubeflow.org/v1beta1"
kind: "InferenceService"
metadata:
  name: "triton-mms"
spec:
  predictor:
    triton:
      args:
      - --log-verbose=1
      resources:
        limits:
          cpu: "1"
          memory: "8Gi"
        requests:
          cpu: "1"
          memory: 8Gi
```

as well as the following TrainedModel

```yaml
apiVersion: "serving.kubeflow.org/v1alpha1"
kind: "TrainedModel"
metadata:
  name: "ourmodelname"
spec:
  inferenceService: triton-mms
  model:
    framework: pytorch
    storageUri: s3:/path-to-container/ourmodelname
    memory: "1Gi"
```

within this s3 container we have stored the following mode:

```
- ourmodelname/
  |_ config.pbtxt
  |_ /1
     |_ model.pt
```

we then observe the following error in the `agent` when loading the mode:
```
{"level":"info","ts":"2021-01-26T08:45:42.264Z","caller":"agent/puller.go:121","msg":"Worker is started for layout"}
{"level":"info","ts":"2021-01-26T08:45:42.264Z","caller":"agent/puller.go:129","msg":"Downloading model from s3:/path-to-container/ourmodelname"}
{"level":"info","ts":"2021-01-26T08:45:42.264Z","caller":"agent/downloader.go:47","msg":"Downloading s3:/path-to-container/ourmodelname to model dir /mnt/models"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x169f489]

goroutine 70 [running]:
github.com/kubeflow/kfserving/pkg/agent/storage.FileExists(0xc0001a4480, 0x7b, 0x1bfab30)
	/go/src/github.com/kubeflow/kfserving/pkg/agent/storage/utils.go:27 +0x99
github.com/kubeflow/kfserving/pkg/agent.(*Downloader).DownloadModel(0xc0003eaa98, 0xc0007c5288, 0x6, 0xc0001a90e0, 0x0, 0x0)
	/go/src/github.com/kubeflow/kfserving/pkg/agent/downloader.go:49 +0x40f
github.com/kubeflow/kfserving/pkg/agent.(*Puller).modelProcessor(0xc0003eaa80, 0xc0007c5288, 0x6, 0xc0001a95c0)
	/go/src/github.com/kubeflow/kfserving/pkg/agent/puller.go:130 +0x273
created by github.com/kubeflow/kfserving/pkg/agent.(*Puller).enqueueModelOp
	/go/src/github.com/kubeflow/kfserving/pkg/agent/puller.go:89 +0x120
```

Interestingly, when we try the same with the example model `cifar10` we do not observe this issue.

**What did you expect to happen:**
Model is loaded successfully 

**Environment:**

- Knative Version: 0.17
- KFServing Version: 0.5.0-rc2

**Comment by yuzliu** [comment]
Thanks! We will take a look.

**Comment by rmporsch** [comment]
It seems we still have issues with this problem after deploying the fix in #1318 .

Previously we had no issue running the cifar10 sample on the documentation. With the newest fix. This seems also broken.

```
{"level":"info","ts":1612776979.9618032,"logger":"fallback-logger","caller":"logging/config.go:78","msg":"Fetch GitHub commit ID from kodata failed","error":"\"KO_DATA_PATH\" does not exist or is empty"}
{"level":"info","ts":1612776979.9618652,"logger":"fallback-logger","caller":"agent/main.go:262","msg":"Initializing model agent with config-dir /mnt/configs, model-dir /mnt/models"}
{"level":"info","ts":1612776979.9618852,"logger":"fallback-logger","caller":"agent/syncer.go:36","msg":"Syncing from model dir /mnt/models"}
{"level":"info","ts":1612776979.9620376,"logger":"fallback-logger","caller":"agent/watcher.go:173","msg":"adding model cifar10"}
{"level":"info","ts":1612776979.9620624,"logger":"fallback-logger","caller":"agent/main.go:271","msg":"Starting puller"}
{"level":"info","ts":1612776979.962122,"logger":"fallback-logger","caller":"agent/watcher.go:89","msg":"Start to watch model config event"}
{"level":"info","ts":1612776979.963104,"logger":"fallback-logger","caller":"agent/puller.go:121","msg":"Worker is started for cifar10"}
{"level":"info","ts":1612776979.9631338,"logger":"fallback-logger","caller":"agent/puller.go:129","msg":"Downloading model from s3://ap-east-1-sup-models/cifar10"}
{"level":"info","ts":1612776979.9632525,"logger":"fallback-logger","caller":"agent/downloader.go:47","msg":"Downloading s3://path_to_bucket/cifar10 to model dir /mnt/models"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x17dd1f2]

goroutine 51 [running]:
github.com/kubeflow/kfserving/pkg/agent.(*Downloader).download(0xc00010fe58, 0xc00085c250, 0x7, 0xc0007477d0, 0x21, 0xc000690090, 0x2)
	/go/src/github.com/kubeflow/kfserving/pkg/agent/downloader.go:86 +0xa2
github.com/kubeflow/kfserving/pkg/agent.(*Downloader).DownloadModel(0xc00010fe58, 0xc00085c250, 0x7, 0xc00007f800, 0x0, 0x0)
	/go/src/github.com/kubeflow/kfserving/pkg/agent/downloader.go:51 +0x4b4
github.com/kubeflow/kfserving/pkg/agent.(*Puller).modelProcessor(0xc00010fe40, 0xc00085c250, 0x7, 0xc000376000)
	/go/src/github.com/kubeflow/kfserving/pkg/agent/puller.go:130 +0x273
created by github.com/kubeflow/kfserving/pkg/agent.(*Puller).enqueueModelOp
	/go/src/github.com/kubeflow/kfserving/pkg/agent/puller.go:89 +0x120
```
